### ISSUE TITLE
[Android] CollectionView selection color not working

### ISSUE THREAD

**Issue by jsuarezruiz** [dotnet/maui#4108]
### Description

CollectionView selection color using **VisualState** not working

```
<Style TargetType="Grid">
    <Setter Property="VisualStateManager.VisualStateGroups">
        <VisualStateGroupList>
            <VisualStateGroup x:Name="CommonStates">
                <VisualState x:Name="Normal" />
                <VisualState x:Name="Selected">
                    <VisualState.Setters>
                        <Setter Property="BackgroundColor" 
                                Value="LightSkyBlue" />
                    </VisualState.Setters>
                </VisualState>
            </VisualStateGroup>
        </VisualStateGroupList>
    </Setter>
</Style>
```

### Steps to Reproduce

1. Download [CollectionViewDemos.zip](https://github.com/dotnet/maui/files/7861241/CollectionViewDemos.zip)
2. Launch the App and navigate to the sample "Vertical list (Selection Color)".

### Version with bug

Preview 11

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

**Comment by shimingsg** [comment]
Verified repro with Android 11 and the above project.

**Comment by HarryLobster** [comment]
This is still an issue in the newest release. How can that be that such a substantial feature is not getting any attention ? I thought that Maui is getting more professional. Seems to me that is still developed by a bunch of developers in their sparetime...

**Comment by Redth** [comment]
![image](https://user-images.githubusercontent.com/271950/166008904-f36be840-7bb6-4e08-a97a-5a3d8e47c09f.png)

Works to select now, however given the list is set to single selection it does not seem that deselecting the item or selecting a different item works in this case.

The other selection demos do work with the platform theming as a workaround for now.

**Comment by VincentBu** [comment]
In 32507.29.main, I can select items in list and the color does change:
![image](https://user-images.githubusercontent.com/44959937/167345969-92169635-974b-4b87-8fa9-da237a5fd0e6.png)

However, as [Redth](https://github.com/Redth) said, deselecting the item doesn't work.

**Comment by AswinPG** [comment]
Is it possible to change the Thickness of Border inside CV as well using VSM's Selected and Normal States ? If it is possible then that's also not working.. Should I add it as a new issue ?

**Comment by rmarinho** [comment]
this was fixed by #7689
